docs/src/content/docs/cookbook/jettons.mdx

```diff
@@ -360,6 +360,48 @@ fun calculateJettonWalletAddress(
 }
 ```
 
+### Onchain metadata creation
+
+```tact
+/// https://github.com/ton-blockchain/TEPs/blob/master/text/0064-token-data-standard.md#jetton-metadata-example-offchain
+fun composeJettonMetadata(
+    name: String,        // full name
+    description: String, // text description of the Jetton
+    symbol: String,      // "stock ticker" symbol without the $ prefix, like USDT or SCALE
+    image: String,       // link to the image
+    // There could be other attributes, see:
+    // https://github.com/ton-blockchain/TEPs/blob/master/text/0064-token-data-standard.md#jetton-metadata-attributes
+): Cell {
+    let dict: map<Int as uint256, Cell> = emptyMap();
+    dict.set(sha256("name"), name.asMetadataCell());
+    dict.set(sha256("description"), description.asMetadataCell());
+    dict.set(sha256("symbol"), symbol.asMetadataCell());
+    dict.set(sha256("image"), image.asMetadataCell());
+
+    return beginCell()
+        .storeUint(0, 8)
+        .storeBit(true)
+        .storeRef(dict.asCell()!!)
+        .endCell();
```

Comment:
Maybe we should serialize it using Tact structs?

It will be smth like
struct MetaCell {
  prefix: Int as uint8; //Should be 0
  data: Cell?; //Maybe Ref - exactly what we need
}

Reply:
Maybe, although seems a bit over the top here and kinda obscures the details. I'd actually replace the `.storeBit(true).storeRef()` with `.storeMaybeRef()` — we have the latter since Tact 1.5.0